features/src/main/scala/com/salesforce/op/features/types/Maps.scala

```diff
@@ -317,7 +317,10 @@ class Prediction private[op](value: Map[String, Double]) extends RealMap(value)
     s"value map must only contain valid keys: '$PredictionName' or " +
       s"starting with '$RawPredictionName' or '$ProbabilityName'"
   )
-  private def keysStartsWith(name: String): Array[String] = value.keys.filter(_.startsWith(name)).toArray.sorted
+
+  // Need to make sure we sort the keys by their final index, which comes after an underscore in the apply function
+  private def keysStartsWith(name: String): Array[String] = value.keys.filter(_.startsWith(name)).toArray
+    .sortBy(_.split('_').last.toInt)
```

Comment:
depending how wide the keys are, but it would be more efficient to scan keys backwards to look for `_` with `lastIndexOf` to map to a `substring`. It would also produce less transient garbage: a single string instead of an array.

Reply:
Too late for the party :)